src/lib/OpenAI/Auth/BaseManifestAuth.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.OpenApi.ApiManifest.OpenAI.Auth;
+using System.Text.Json;
+
+namespace Microsoft.OpenApi.ApiManifest.OpenAI;
+
+public abstract class BaseManifestAuth
```

Comment:
we probably want to avoid abbreviations in public type names, thoughts?

Reply:
+1 to not using abbreviations where possible. In this case, `BaseManifestAuth` is how OpenAI names their authentication interface - https://platform.openai.com/docs/plugins/getting-started/plugin-manifest.

Reply:
Although I feel like the namespace should be "Authentication" instead of "Auth" for consistency with their TOC - https://platform.openai.com/docs/plugins/authentication. Thoughts, @baywet?